File: ts-sdk/src/sdk/models/operations/getcollaborationwhitelistentriesid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCollaborationWhitelistEntriesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=collaboration_whitelist_entry_id" })
  collaborationWhitelistEntryId: string;
}


export class GetCollaborationWhitelistEntriesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetCollaborationWhitelistEntriesIdPathParams;
}


export class GetCollaborationWhitelistEntriesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  collaborationAllowlistEntry?: shared.CollaborationAllowlistEntry;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/getcollaborationwhitelistentriesid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetCollaborationWhitelistEntriesIDPathParams struct {
	CollaborationWhitelistEntryID string `pathParam:"style=simple,explode=false,name=collaboration_whitelist_entry_id"`
}

type GetCollaborationWhitelistEntriesIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetCollaborationWhitelistEntriesIDPathParams
}

type GetCollaborationWhitelistEntriesIDResponse struct {
	ClientError                 *shared.ClientError
	CollaborationAllowlistEntry *shared.CollaborationAllowlistEntry
	ContentType                 string
	StatusCode                  int64
}
